src/main/java/com/researchspace/service/impl/RecordManagerImpl.java

```diff
@@ -1222,16 +1215,16 @@ private List<BaseRecord> getRecordsSharedWithUser(User u, boolean onlyTemplates,
         recordGroupSharingDao.getSharedRecordsWithUser(u), PermissionType.READ, subject);
   }
 
-  private Set<BaseRecord> getViewableRecordsByUser(User u, boolean onlyTemplates) {
+  private Set<BaseRecord> getViewableRecordsByUsers(List<User> users, boolean onlyTemplates) {
     if (onlyTemplates) {
-      return recordDao.getViewableTemplatesForUser(u);
+      return recordDao.getViewableTemplatesForUsers(users);
     }
-    return recordDao.getViewableRecordsForUser(u);
+    return recordDao.getViewableRecordsForUsers(users);
   }
 
   @Override
-  public Set<BaseRecord> getViewableTemplates(User user) {
-    return recordDao.getViewableTemplatesForUser(user);
+  public Set<BaseRecord> getViewableTemplates(List<User> users) {
```

Comment:
What do you think about having Set<User>`? 
I mean if we put all in the `Set` (for the `Users` that are always uniquely treated) all the retainAll and join operations will be very efficient rather tha using the `List`s